src/client/fuse_ll.cc

```diff
@@ -163,6 +168,101 @@ class CephFuse::Handle {
   struct fuse_args args;
 };
 
+static bool already_mounted(const char *path)
+{
+  struct stat orig_stat;
+  struct stat parent_stat;
+  char *orig_path = NULL;
+  char *parent_path = NULL;
+  bool ret_val = false;
+  int err = 0;
+
+  do
+  {
+    orig_path = new char[strlen(path) + 1];
+    memcpy(orig_path, path, strlen(path) + 1);
+    parent_path = dirname(orig_path);
+
+    // get stat information for original path
+    err = stat(path, &orig_stat);
```

Comment:
It's better to use statx(2) in new code.

Old-school stat ends up fetching a bunch of fields that you don't care about here. For most filesystems it doesn't matter much, but for ceph, it means requesting more cap access than you really need. I'd suggest just doing a statx on both with a mask that just has STX_INO and then compare the stx_dev_* fields in there.

Reply:
Oh, and you should also be able to use AT_STATX_DONT_SYNC, since this info is not subject to change on the MDS.

Reply:
Done